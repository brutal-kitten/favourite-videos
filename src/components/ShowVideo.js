import React, { Component } from 'react';
import YouTube from 'react-youtube';


class ShowVideo extends Component {


  render () {

    const opts = {
      height: '390',
      width: '640',
      playerVars: {
        autoplay: 1
      }
    };

    return (
      <YouTube
        videoId={this.props.videoId}
        opts={opts}
        onReady={this._onReady}
      />
    );
  }

  _onReady(event) {
    event.target.pauseVideo();
  }
}

export default ShowVideo
