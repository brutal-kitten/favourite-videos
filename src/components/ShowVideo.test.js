import React from 'react';
import ReactDOM from 'react-dom';
import ShowVideo from './ShowVideo.js';
import TestRenderer from 'react-test-renderer';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

it('renders properly', () => {
  const tree = TestRenderer.create(
    <ShowVideo />).toJSON();
  expect(tree).toMatchSnapshot();
});

const mockCloseModal = jest.fn();
const props = {
  videoId: "lalala",
  closeModal: mockCloseModal
}

const wrapper = shallow(<ShowVideo {...props}/>);

wrapper.find('button.closeModal').simulate('click', {
  preventDefault: () => {},
});

it('should call function closeModal once', () => {
  expect(mockCloseModal.mock.calls.length).toBe(1);
});

it('should call mockPauseVideo once', () => {
  const mockPauseVideo = jest.fn();
  wrapper.instance()._onReady({ target: {
    pauseVideo: mockPauseVideo
  }});
  expect(mockCloseModal.mock.calls.length).toBe(1);
});
