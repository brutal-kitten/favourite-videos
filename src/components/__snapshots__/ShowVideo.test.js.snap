// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<div
  className="window"
>
  <button
    className="closeModal"
    onClick={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="glyphicon glyphicon-remove"
    />
  </button>
  <span
    className=""
  >
    <div
      className={null}
      id={null}
    />
  </span>
</div>
`;
